app/src/src_nonfreenet/org/breezyweather/sources/meteoam/MeteoAmResultConverter.kt

```diff
@@ -0,0 +1,213 @@
+        || params[4] != "wdir" || params[7] != "wkmh" || params[9] != "icon") {
+        throw InvalidOrIncompleteDataException()
+    }
+
+    if (oParams.size < 9 || oParams[0] != "2t" || oParams[1] != "r" || oParams[2] != "pmsl"
+        || oParams[3] != "wdir" || oParams[6] != "wkmh" || oParams[8] != "icon") {
+        throw InvalidOrIncompleteDataException()
+    }
+
+    return WeatherWrapper(
+        current = if (observation != null) getCurrent(observation) else null,
+        dailyForecast = getDailyForecast(stats),
+        hourlyForecast = getHourlyForecast(timeseries, data)
+    )
+}
+
+private fun getCurrent(
+    currentResult: MeteoAmObservationDatasets
+): Current {
+    return Current(
+        weatherText = getWeatherText(currentResult.icon?.get("0")),
+        weatherCode = getWeatherCode(currentResult.icon?.get("0")),
+        temperature = Temperature(
+            currentResult.temp?.get("0")?.toDouble()
+        ),
+        wind = Wind(
+            degree = when (currentResult.wdir?.get("0")) {
+                is Double -> currentResult.wdir["0"] as Double
+                "VRB" -> -1.0
+                else -> null
+            },
+            speed = when (currentResult.wkph?.get("0")) {
+                is Double -> currentResult.wkph["0"] as Double / 3.6
+                else -> null
+            }
+        ),
+        relativeHumidity = currentResult.rhum?.get("0")?.toDouble(),
+        pressure = currentResult.pres?.get("0")?.toDouble()
+    )
+}
+
+private fun getDailyForecast(
+    dailyResult: List<MeteoAmForecastStats>
+): List<Daily> {
+    val formatter = SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ssX", Locale.ITALIAN)
+    formatter.timeZone = TimeZone.getTimeZone("Europe/Rome")
+    val dailyForecast = mutableListOf<Daily>()
+    dailyResult.forEach {
+        if (it.icon != "-") {
+            dailyForecast.add(
+                Daily(
+                    date = formatter.parse(it.localDate)!!,
+                    day = HalfDay(
+                        weatherText = getWeatherText(it.icon),
+                        weatherCode = getWeatherCode(it.icon)
+                    ),
+                    night = HalfDay(
+                        weatherText = getWeatherText(it.icon),
+                        weatherCode = getWeatherCode(it.icon)
+                    )
+                )
+            )
+        }
+    }
+    return dailyForecast
+}
+
+private fun getHourlyForecast(
+    timeseries: List<String>,
+    data: MeteoAmForecastData
+): List<HourlyWrapper> {
+    val formatter = SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ssX", Locale.ITALIAN)
+    formatter.timeZone = TimeZone.getTimeZone("Europe/Rome")
+    val hourlyForecast = mutableListOf<HourlyWrapper>()
+    for (i in timeseries.indices) {
+        hourlyForecast.add(
+            HourlyWrapper(
+                date = formatter.parse(timeseries[i])!!,
+                weatherText = getWeatherText(data.icon?.get(i.toString())),
```

Comment:
Just to be safe:
```kt
getOrElse(i.toString()) { null }
```

Do the same for every others below

Reply:
Fixed.